src/main/java/aztech/modern_industrialization/machines/multiblocks/HatchType.java

```diff
@@ -23,42 +23,40 @@
  */
 package aztech.modern_industrialization.machines.multiblocks;
 
-import aztech.modern_industrialization.MI;
-import aztech.modern_industrialization.MIText;
 import com.mojang.datafixers.util.Either;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.resources.ResourceLocation;
 
-public enum HatchType {
-    ITEM_INPUT(0, MIText.ItemInputHatch),
-    ITEM_OUTPUT(1, MIText.ItemOutputHatch),
-    FLUID_INPUT(2, MIText.FluidInputHatch),
-    FLUID_OUTPUT(3, MIText.FluidOutputHatch),
-    ENERGY_INPUT(4, MIText.EnergyInputHatch),
-    ENERGY_OUTPUT(5, MIText.EnergyOutputHatch),
-    NUCLEAR_ITEM(6, "nuclear_item_hatch"),
-    NUCLEAR_FLUID(7, "nuclear_fluid_hatch"),
-    LARGE_TANK(8, "large_tank_hatch");
+public class HatchType {
+    private final ResourceLocation id;
+    private final Either<MutableComponent, ResourceLocation> descriptionOrBlockId;
```

Comment:
I would probably use a normal `Component` and copy it in `description`, to make sure that the mutable component doesn't get modified here.